### ISSUE TITLE
Architecture Plan

### ISSUE THREAD

**Issue by PTNobel** [udellgroup/rlaopt#59]
- [ ] Lasso and matrix completion example
  - [ ] Make atoms
    - [ ] $\ell_2^2$ norm
    - [ ] $\ell_1$ norm 
    - [ ] $\Vert\cdot\Vert_*$ atom
    - [ ] Affine expression
  - [x] Create OperatorSplit composer
  - [ ] Port SURE-CR Accel Prox Grad

Linear Operators
- [ ] Port in `torch_linops`?

Methods for `OperatorSplit` and atoms (for class hierachy of atoms see CVXPY)
- Should be a subclass of `torch.nn.Module`
- [x] `__add__`: Base on RandALO/modeling_layer.py to implement Sum class
- [x] `to_cvxpy` for testing: Base on RandALO/model_layer.py
- [x] `smooth`/`gradient`
- [ ] `proxable`/`prox`
- [ ] `subsampleable`/`subsample` (pass in mask, returns masked object)

Solver methods:
 - Should be a subclass of `torch.autograd.Function`
 - [ ] `forward`
 - [ ] `backward`
